sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1278,10 +1278,16 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with Logging {
   }
 
   /** Create a numeric literal expression. */
-  private def numericLiteral(ctx: NumberContext)(f: String => Any): Literal = withOrigin(ctx) {
-    val raw = ctx.getText
+  private def numericLiteral(ctx: NumberContext)(range: (BigDecimal, BigDecimal), typeName: String)
```

Comment:
a small nit - maybe we should just take minValue, maxValue , typeName? rather than putting min value and max value in a tuple.